src/AccordionWithHeader.js

```diff
@@ -60,6 +60,17 @@ export default class AccordionWithHeader extends Component {
     }
   }
 
+  componentDidUpdate() {
+    const hasNewChildren =  this.props.children.length > this.state.children.length;
```

Comment:
Should the panels update when a child is removed as well?

Reply:
Probably! Ideally this check would be inclusive of adding AND subtracting panels, while maintaining the Open state of the current panel.  

For example, if I have 5 panels, and number 2 is open, if I remove number 3, then number 2 should stay open 😁 . Same with adding one, as `hasNewChildren` demonstrates above. 

So, perhaps something like this would make more sense from a logic standpoint: 

```js
const hasUpdatedCount =  this.props.children.length !== this.state.children.length;
```